bundles/framework/admin-announcements/instance.js

```diff
@@ -0,0 +1,215 @@
+        init: function () {
+            return null;
+        },
+        /**
+         * @method update
+         * implements BundleInstance protocol update method - does nothing atm
+         */
+        update: function () {
+
+        },
+
+        /**
+         * @method stop
+         * implements BundleInstance protocol stop method
+         */
+        stop: function () {
+            const sandbox = this.sandbox();
+
+            const request = Oskari.requestBuilder('userinterface.RemoveExtensionRequest')(this);
+            sandbox.request(this, request);
+
+            this.sandbox.unregister(this);
+            this.started = false;
+        },
+        /**
+         * @method startExtension
+         * implements Oskari.userinterface.Extension protocol startExtension method
+         * Creates a flyout and a tile:
+         * Oskari.framework.bundle.admin-announcements.Flyout
+         * Oskari.framework.bundle.admin-announcements.Tile
+         */
+        startExtension: function () {
+            this.plugins['Oskari.userinterface.Flyout'] = Oskari.clazz.create(
+                'Oskari.framework.bundle.admin-announcements.Flyout', this);
+            this.plugins['Oskari.userinterface.Tile'] = Oskari.clazz.create(
+                'Oskari.framework.bundle.admin-announcements.Tile', this, this.getLocalization());
+        },
+        /**
+         * @method stopExtension
+         * implements Oskari.userinterface.Extension protocol stopExtension method
+         * Clears references to flyout and tile
+         */
+        stopExtension: function () {
+            this.plugins['Oskari.userinterface.Flyout'] = null;
+            this.plugins['Oskari.userinterface.Tile'] = null;
+        },
+        /**
+         * @method getPlugins
+         * implements Oskari.userinterface.Extension protocol getPlugins method
+         * @return {Object} references to flyout and tile
+         */
+        getPlugins: function () {
+            return this.plugins;
+        },
+        /**
+         * @method getTitle
+         * @return {String} localized text for the title of the component
+         */
+        getTitle: function () {
+            return this.getLocalization('title');
+        },
+        /**
+         * @method getDescription
+         * @return {String} localized text for the description of the component
+         */
+        getDescription: function () {
+            return this.getLocalization('desc');
+        },
+        /**
+         * @method createUi
+         * (re)creates the UI for admin-announcements functionality
+         */
+        createUi: function () {
+            this.plugins['Oskari.userinterface.Tile'].refresh();
+        }
+    }, {
+        /**
+         * @property {String[]} protocol
+         * @static
+         */
+        protocol: [
+            'Oskari.bundle.BundleInstance',
+            'Oskari.mapframework.module.Module',
+            'Oskari.userinterface.Extension'
+        ]
```

Comment:
Most of the code here is unnecessary boilerplate that could be removed if you just would extend the DefaultExtension instead of defining these protocols:

        extend: ['Oskari.userinterface.extension.DefaultExtension']

I just realized that this is true for the new layerlist bundle as well. But "this is the way" and all that.

Reply:
extend now used in instances and flyouts